robot/src/main/java/us/ilite/robot/modules/Elevator.java

```diff
@@ -0,0 +1,280 @@
+        if (pCurrentDirectionUp) {
+            if (pCurrentEncoderTicks >= mUpEncoderThreshold) {
+                mCurrentState = EElevatorState.DECEL_TOP;
+            }
+        } else if (!pCurrentDirectionUp) {
+            if (pCurrentEncoderTicks >= mDownEncoderThreshold) {
+                mCurrentState = EElevatorState.DECEL_BOTTOM;
+            }
+        }
+    }
+    
+    public void toBottom() {
+        mDesiredPosition = EElevatorPosition.BOTTOM;
+        mCurrentState = EElevatorState.SET_POSITION;
+    }
+
+    public void toTop() {
+        mDesiredPosition = EElevatorPosition.BOTTOM;
+        mCurrentState = EElevatorState.SET_POSITION;
+    }
+
+
+    private double setPosition(EElevatorPosition pDesiredPosition, double pCurrentTime) {
+    
+
+        double power;
+        //If the current encoder ticks is less than the threshold of the desired position, then
+        //The desired position is above the initial position
+        mDesiredPositionAboveInitial = (mCurrentEncoderTicks < pDesiredPosition.kEncoderThreshold);
+        
+        
+
+        //If the we are under the desired position and we are not in the encoder position for said
+        //position, then we are still setting position
+        if (mDesiredPositionAboveInitial && mCurrentEncoderTicks >= pDesiredPosition.kEncoderThreshold) {
+            mSettingPosition = false;
+        } else if(!mDesiredPositionAboveInitial && mCurrentEncoderTicks <= pDesiredPosition.kEncoderThreshold) { //if we are above it, same logic
+            mSettingPosition = false;
+        } else {
+            mCurrentPosition = pDesiredPosition;
+            mSettingPosition = true;
+        }
+
+        //If we make it to or beyond our position, then hold.
+        if (!mSettingPosition) {
+            power = EElevatorState.HOLD.getPower();
+        } else {
+            mPidController.setSetpoint(pDesiredPosition.kEncoderThreshold); //Our set point is the threshold of the destination state
+            power = mPidController.calculate(mCurrentEncoderTicks, pCurrentTime);
+            power = Util.limit(mDesiredPower, mMinPower, mMaxPower); //limit power from -1.0 - 1.0
+        }
+
+        return power;
+ 
+    }
+
+    public int getCurrentEncoderTicks() {
+        return mCurrentEncoderTicks;
+    }
+
+    public void setPower( double pPower ) {
+        mDesiredPower = pPower;
+    }
+
+    public boolean getDirection() {
+        return mDesiredDirectionUp;
+    }
+
+    public double getDesiredPower() {
+        return mDesiredPower;
+    }
+
+
+    //Getting voltage and current
+    public double getBusVoltage() {
+        return mMasterElevator.getBusVoltage();
+    }
+
+    public double getMasterCurrent() {
+        return mMasterElevator.getOutputCurrent();
+    }
+
+    public boolean finishedPositioning() {
+        return !mSettingPosition;
+    }
+
+    public int getEncoderPosition() {
+        return mEncoder.getSelectedSensorPosition(0);
+    }
+
```

Comment:
This is dangerous.  This means outside methods can directly influence the encoder position, meaning we lose that control and visibility inside this class.  This method should be private, or removed.